askama_derive/src/config.rs

```diff
@@ -168,24 +168,29 @@ impl<'a> TryFrom<RawSyntax<'a>> for Syntax<'a> {
             comment_end: raw.comment_end.unwrap_or(default.comment_end),
         };
 
-        if syntax.block_start.len() != 2
-            || syntax.block_end.len() != 2
-            || syntax.expr_start.len() != 2
-            || syntax.expr_end.len() != 2
-            || syntax.comment_start.len() != 2
-            || syntax.comment_end.len() != 2
-        {
-            return Err("length of delimiters must be two".into());
+        for s in [
+            syntax.block_start,
+            syntax.block_end,
+            syntax.expr_start,
+            syntax.expr_end,
+            syntax.comment_start,
+            syntax.comment_end,
+        ] {
+            if s.len() < 2 {
+                return Err("Delimiters must be at least two characters long".into());
+            } else if s.chars().any(|c| c.is_whitespace()) {
+                return Err("Delimiters may not contain white spaces".into());
+            }
         }
 
-        let bs = syntax.block_start.as_bytes()[0];
-        let be = syntax.block_start.as_bytes()[1];
-        let cs = syntax.comment_start.as_bytes()[0];
-        let ce = syntax.comment_start.as_bytes()[1];
-        let es = syntax.expr_start.as_bytes()[0];
-        let ee = syntax.expr_start.as_bytes()[1];
-        if !((bs == cs && bs == es) || (be == ce && be == ee)) {
-            return Err(format!("bad delimiters block_start: {}, comment_start: {}, expr_start: {}, needs one of the two characters in common", syntax.block_start, syntax.comment_start, syntax.expr_start).into());
+        for (s1, s2) in [
+            (syntax.block_start, syntax.expr_start),
+            (syntax.block_start, syntax.comment_start),
+            (syntax.expr_start, syntax.comment_start),
+        ] {
+            if s1.starts_with(s2) || s2.starts_with(s1) {
+                return Err("A delimiter may not be the prefix of another delimiter".into());
```

Comment:
Nit: I think our error message usually don't start with capitals.

Reply:
You're right. Will change.